app/javascript/react/components/SegmentedPickerButtons/SegmentedPickerButtons.style.tsx

```diff
@@ -0,0 +1,38 @@
+import styled from "styled-components";
+
+import { blue, lightGray } from "../../assets/styles/colors";
+
+const SegmentedPickerContainer = styled.div`
+  position: relative;
+  display: inline-flex;
+  border: 1px solid ${lightGray};
+  border-radius: 5px;
+`;
+
+interface SegmentedOptionProps {
+  isSelected: boolean;
+  isFirst: boolean;
+  isLast: boolean;
+}
+
+const SegmentedOption = styled.div<SegmentedOptionProps>`
```

Comment:
The given approach shows that our `SegmentedPicker` does need to know what means `first` and `last`. There is no need for this, as only style needs this. I would personally pass high-level info and decompose it later. I would like to suggest something below. One more thing is the conditional `border-radius` a bit too complicated I think 😅

NOI => Number of items

```
interface SegmentedOptionProps {
  isSelected: boolean;
  index: number;
  NOI: number;
}

interface Border {
  index: number;
  NOI: number;
}

const getBorder = ({ index, NOI }: Border) => {
  const leftBorder = "5px 0 0 5px"
  const rightBorder = "0px 5px 5px 0px"

  if (index == NOI - 1) {
    return rightBorder
  } else if (index == 0) {
    return leftBorder
  }
}

const SegmentedOption = styled.div<SegmentedOptionProps>`
  padding: 8px 15px;
  margin: 2px;
  cursor: pointer;
  background-color: ${({ isSelected }) => (isSelected ? blue : "transparent")};
  border-radius: ${({ isSelected, index, NOI }) => isSelected && getBorder({index, NOI})};
  color: ${({ isSelected }) => (isSelected ? blue : "#fff")};
  > span {
    color: ${({ isSelected }) => (isSelected ? "#fff" : lightGray)};
    text-transform: uppercase;
  }
`;




<S.SegmentedOption
          key={option}
          onClick={() => onOptionChange(option)}
          isSelected={option === selectedOption}
          index={index}
          NOI={options.length}
        >
          <span>{option}</span>
        </S.SegmentedOption>
```

Reply:
Good point, thanks <3